### PR TITLE
Hometask 8

### PR DESCRIPTION
**Pull request by Saidabdulla**
Separated to components ✅

Question:
In main page I have accounts section and account cards, I have two options while working with arrays.
1) To create component for card and and use ngFor while using app-card in main component, like this:

<main-cards ngFor.....>
    <app-card   [dataForCard]="data" > </app-card>
</main-cards>

Is it ok to call component many times?

2) Create component of all cards with their parent and give all data (array from DB) to rightly inside component and use for here:

which is right way to do this?
Now i am using second way.

### CODE REVIEW

**Comment by Morphine-medicine** [comment]
> Separated to components ✅
> 
> Question: In main page I have accounts section and account cards, I have two options while working with arrays.
> 
> 1. To create component for card and and use ngFor while using app-card in main component, like this:
> 
> Is it ok to call component many times?
> 
> 2. Create component of all cards with their parent and give all data (array from DB) to rightly inside component and use for here:
> 
> which is right way to do this? Now i am using second way.

Sory for delay, I'll recommend to use *ngFor here

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { Router } from '@angular/router';
 })
 export class AuthFormComponent {
     errorMessage = '';
+    hide = true;
```
Comment (Morphine-medicine) [inline client/src/app/auth/auth-form/auth-form.component.ts:13]: I'll recommend you to name this like isPasswordVisible

File: client/src/app/layout/header/header.component.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+})
+export class HeaderComponent {
+    @Input() isLoggedIn!: boolean;
+    @Output() menuButtonClicked: EventEmitter<void> = new EventEmitter<void>();
+    @Output() loggedOutClicked: EventEmitter<void> = new EventEmitter<void>();
+    constructor() {}
+
+    onClick(): boolean {
+        this.loggedOutClicked.emit();
+        return false;
```
Comment (Morphine-medicine) [inline client/src/app/layout/header/header.component.ts:16]: Not sure that returned value is needed, so here you can use void

**Review by Morphine-medicine — APPROVED** [review]
Mostly looks good, take a look on comments and then merge without additional review

**Comment by Saidabdulla** [comment]
Understand, Thanks.

**Pull request merged**